cmd/podman/common/netflags.go

```diff
@@ -185,37 +192,41 @@ func NetFlagsToNetOptions(opts *entities.NetOptions, flags pflag.FlagSet) (*enti
 		opts.Networks = networks
 	}
 
-	if flags.Changed("ip") || flags.Changed("mac-address") || flags.Changed("network-alias") {
+	if flags.Changed("ip") || flags.Changed("ip6") || flags.Changed("mac-address") || flags.Changed("network-alias") {
 		// if there is no network we add the default
 		if len(opts.Networks) == 0 {
 			opts.Networks = map[string]types.PerNetworkOptions{
 				"default": {},
 			}
 		}
 
-		ip, err := flags.GetString("ip")
-		if err != nil {
-			return nil, err
-		}
-		if ip != "" {
-			// if pod create --infra=false
-			if infra, err := flags.GetBool("infra"); err == nil && !infra {
-				return nil, errors.Wrap(define.ErrInvalidArg, "cannot set --ip without infra container")
+		for _, ipFlagName := range []string{"ip", "ip6"} {
+			ip, err := flags.GetString(ipFlagName)
+			if err != nil {
+				return nil, err
 			}
+			if ip != "" {
+				// if pod create --infra=false
+				if infra, err := flags.GetBool("infra"); err == nil && !infra {
+					return nil, errors.Wrapf(define.ErrInvalidArg, "cannot set --%s without infra container", ipFlagName)
+				}
 
-			staticIP := net.ParseIP(ip)
-			if staticIP == nil {
-				return nil, errors.Errorf("%s is not an ip address", ip)
-			}
-			if !opts.Network.IsBridge() && !opts.Network.IsDefault() {
-				return nil, errors.Wrap(define.ErrInvalidArg, "--ip can only be set when the network mode is bridge")
-			}
-			if len(opts.Networks) != 1 {
-				return nil, errors.Wrap(define.ErrInvalidArg, "--ip can only be set for a single network")
-			}
-			for name, netOpts := range opts.Networks {
-				netOpts.StaticIPs = append(netOpts.StaticIPs, staticIP)
-				opts.Networks[name] = netOpts
+				staticIP := net.ParseIP(ip)
+				if staticIP == nil {
+					return nil, errors.Errorf("%q is not an ip address", ip)
+				}
+				// Do we want to validate that --ip only accepts ipv4 and --ip6 only accepts ipv6 addresses?
+				// There is no difference in the backend anyway so I would say no.
```

Comment:
Please remove this comment before final merge.  I'd say do what Docker does here.